### PR TITLE
Minor: move `Column` related tests and rename `column.rs`

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?
N/A

## Rationale for this change
These tests belong to `Column`. It would be better to put them together.

## What changes are included in this PR?
Move tests

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/physical-expr/src/expressions/column.rs [hunk 1]
```diff
@@ -100,49 +100,3 @@ impl PartialEq<dyn Any> for UnKnownColumn {
         false
     }
 }
-
```
Comment (alamb) [inline datafusion/physical-expr/src/expressions/column.rs:103]: It seems like `datafusion/physical-expr/src/expressions/column.rs` could also be renamed to `datafusion/physical-expr/src/expressions/unknown_column.rs` or `datafusion/physical-expr/src/expressions/unknown.rs` to better match its contents

**Review by alamb — APPROVED** [review]
Looks good to me -- thanks @jonahgao

File: datafusion/physical-expr/src/expressions/column.rs (hunk 1 above)
Comment (jonahgao) [reply datafusion/physical-expr/src/expressions/column.rs:103]: Renamed it to `unknown_column.rs` in https://github.com/apache/datafusion/pull/11573/commits/f4407ac092ee639ea63dd807cae0f68f49ec0d62

**Comment by alamb** [comment]
🚀

**Pull request merged**